op-challenger/game/fault/preimages/split.go

```diff
@@ -0,0 +1,33 @@
+package preimages
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+)
+
+var _ PreimageUploader = (*SplitPreimageUploader)(nil)
+
+// SplitPreimageUploader routes preimage uploads to the appropriate uploader
+// based on the size of the preimage.
+type SplitPreimageUploader struct {
+	directUploader PreimageUploader
+	largeUploader  PreimageUploader
+}
+
+func NewSplitPreimageUploader(directUploader PreimageUploader, largeUploader PreimageUploader) *SplitPreimageUploader {
+	return &SplitPreimageUploader{directUploader, largeUploader}
+}
+
+func (s *SplitPreimageUploader) UploadPreimage(ctx context.Context, parent uint64, data *types.PreimageOracleData) error {
+	if data != nil {
+		// TODO(proofs#467): determine the correct size threshold to toggle between
+		//                   the direct and large preimage uploaders.
+		if len(data.OracleData) > 32 {
```

Comment:
nit: Probably worth declaring this as a constant (I'm not a strict no magic numbers believer but this feels like something that would benefit from having a name).

Reply:
oh actually, 32 bytes is way too small.  Suggest setting this to something very large initially until we know the actual limit and have a working large preimage uploader.